spec/policy/maintenance_mode/maintenance_mode_spec.lua

```diff
@@ -1,8 +1,15 @@
 local MaintenancePolicy = require('apicast.policy.maintenance_mode')
+local ngx_variable = require('apicast.policy.ngx_variable')
 
 describe('Maintenance mode policy', function()
-  describe('.rewrite', function()
+
+  describe('.access', function()
     before_each(function()
+      ctx = { service = { api_backend = 'http://backend2.example.org:80' } }
+      ctx2 = {
```

Comment:
does this mean we have 2 different instances of the context object during runtime? that could be problematic no as from anywhere we only ever reference it as `context` or `ctx`

Reply:
my intention here was to use a different context (ctx2) just for some of the tests. I guess I can also achieve this by modifying the same ctx object from within the test where I need a different value.

Reply:
fixed this as per our discussion, I also added a couple more tests.